src/scripts/main.js

```diff
@@ -1,3 +1,276 @@
 'use strict';
 
-// write code here
+const thead = document.querySelector('tr');
+const tBody = document.querySelector('tbody');
+let rowsList = tBody.querySelectorAll('tr');
+
+const leaveNumbers = new RegExp(/[^\d]/g);
+
+function numbersFilter(value) {
+  return Number(value.replace(leaveNumbers, ''));
+}
+
+function sortByColumnName(
+  callback, tableRows, columnIndex, fieldName, isColumnReversed
+) {
+  const sortStrings = (prevPerson, nextPerson) => {
+    const prevProperty = prevPerson.children[columnIndex].textContent;
+    const nextProperty = nextPerson.children[columnIndex].textContent;
+
+    return isColumnReversed
+      ? prevProperty.localeCompare(nextProperty)
+      : nextProperty.localeCompare(prevProperty);
+  };
+
+  const sortNumbers = (prevPerson, nextPerson) => {
+    const prevProperty = prevPerson.children[columnIndex].textContent;
+    const nextProperty = nextPerson.children[columnIndex].textContent;
+
+    if (prevProperty.match(leaveNumbers) !== null) {
+      return isColumnReversed
+        ? callback(prevProperty) - callback(nextProperty)
+        : callback(nextProperty) - callback(prevProperty);
+    }
+
+    return isColumnReversed
+      ? prevProperty - nextProperty
+      : nextProperty - prevProperty;
+  };
+
+  let sortedColumn = null;
+
+  switch (fieldName) {
+    case 'Name':
+    case 'Position':
+    case 'Office':
+      sortedColumn = [...tableRows].sort(sortStrings);
+      break;
+
+    case 'Age':
+    case 'Salary':
+      sortedColumn = [...tableRows].sort(sortNumbers);
+      break;
+
+    default:
+      break;
+  }
+
+  return sortedColumn;
+}
+
+let previousHeadline;
+let isReversed = true;
+
+thead.addEventListener('click', ckickEvent => {
+  ckickEvent.preventDefault();
+
+  const currentHeadline = ckickEvent.target.textContent;
+
+  if (previousHeadline === currentHeadline) {
+    isReversed = !isReversed;
+  } else {
+    isReversed = true;
+  }
```

Comment:
тут что то не понятное, по идеи должно работать так
```suggestion
  if (previousHeadline === currentHeadline) {
    isReversed = true;
  } else {
    isReversed = false;
  }
```